apps/omg_watcher/lib/omg_watcher/exit_processor/core.ex

```diff
@@ -366,7 +366,9 @@ defmodule OMG.Watcher.ExitProcessor.Core do
   end
 
   @spec finalize_in_flight_exits(t(), [map()]) ::
-          {:ok, t(), list()} | {:unknown_piggybacks, list()} | {:unknown_in_flight_exit, MapSet.t(non_neg_integer())}
+          {:ok, t(), list(), list(), list()}
```

Comment:
let's put a `doc` here explaining what the args and returns are. This just got way more complicated.

**Among other** explanations, make sure to include what the kinds of lists are the two new lists - that they "fit" to whatever `State.Core` likes to receive in `exit_utxos` while handling IFEs.

Reply:
Next, add tests under `state_interaction_test.exs`, that check that whatever you're feeding `State.Core` makes sense.

You can even add a test, where some IFE finalizes invalidly and check if you're getting the Logger warning for completeness (`ExUnit.CaptureLog`). Later we'd swap this test for test of more proper handling.

This combined should give you decent test coverage of the IFE finalization